pandas/tests/io/conftest.py

```diff
@@ -184,3 +185,29 @@ def add_tips_files(bucket_name):
     while cli.list_buckets()["Buckets"] and timeout > 0:
         time.sleep(0.1)
         timeout -= 0.1
+
+
+_compression_formats_params = [
+    (".no_compress", None),
+    ("", None),
+    (".gz", "gzip"),
+    (".GZ", "gzip"),
+    (".bz2", "bz2"),
+    (".BZ2", "bz2"),
+    (".zip", "zip"),
+    (".ZIP", "zip"),
+    (".xz", "xz"),
+    (".XZ", "xz"),
+    pytest.param((".zst", "zstd"), marks=td.skip_if_no("zstandard")),
+    pytest.param((".ZST", "zstd"), marks=td.skip_if_no("zstandard")),
```

Comment:
If the compression tests take too much time: I would expect that we need only one upper-case test, we have one `.lower()` call that affects all compression formats. @jreback

Reply:
Small benchmark: `io/test_common.py` + `io/test_pickle.py` with full list: 4.25 s, with small list: 4.2 s

Reply:
Thank you for the quick benchmark! Based on that we can keep all the upper cases. Compression is also used in a lot of CSV tests.